mypy/util.py

```diff
@@ -134,3 +141,54 @@ def id(self, o: object) -> int:
             self.id_map[o] = self.next_id
             self.next_id += 1
         return self.id_map[o]
+
+
+# default timeout is short in case this functions is called individually for many files
+# batch processing results in better performance
+def wait_for(funcs: Iterable[Callable[[], None]],
+             exc: Iterable[Type[BaseException]] = (),
+             timeout: float = 0.1,
```

Comment:
I agree, but now that I simplified the whole thing to get rid of `wait_for`, our only API is `util.replace`. I am guessing we should keep `timeout` there as an optional last param, right?

Btw, do you prefer the simplified (current) version where `wait_for` doesn't exist as a standalone function?